In [1]:
# Importing Libs
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from collections import Counter
# My Tools
%matplotlib inline

In [2]:
# Reseting figure size params.
plt.rcParams['figure.figsize'] = [12, 7]

In [3]:
# Setting seaborn as default plotting lib.
sns.set()
sns.set_theme(palette='pastel')

In [4]:
# Reading data.
df = pd.read_csv('encuesta_baches.csv')

## Showing df.

In [5]:
df.head()

Marca temporal Lugar de residencia.  \
0  29/10/2022 21:02:13            Venezuela   
1   30/10/2022 9:28:12            Argentina   
2  30/10/2022 10:05:24               Mexico   
3  30/10/2022 10:12:40                Spain   
4  30/10/2022 10:21:56            Singapore   

  Si selecciono PARAGUAY como lugar de residencia. Puede indicar el departamento en el que vive actualmente.  \
0                                                NaN                                                           
1                                                NaN                                                           
2                                                NaN                                                           
3                                                NaN                                                           
4                                            Central                                                           

  Como calificaría el estado de las calles en su lugar de residencia.  \
0                                               Malo                    
1                                              Bueno                    
2                                         Lamentable                    
3                                              Bueno                    
4                                           Exelente                    

  Considera que exista un problema de baches en su lugar de residencia.  \
0                                                 SI                      
1                                                 SI                      
2                                                 SI                      
3                                                 SI                      
4                                                 NO                      

  Puede hacer una estimación de la cantidad de baches en su lugar de residencia.  \
0                                             MUCHOS                               
1                                              POCOS                               
2                                             MUCHOS                               
3                                              POCOS                               
4                                                NaN                               

  Tipo de camino en su lugar de residencia.  \
0                                   ASFALTO   
1                                   ASFALTO   
2                                   ASFALTO   
3                                   ASFALTO   
4                                   ASFALTO   

  Cuanto tiempo cree que demora la reparación de baches en su lugar de residencia.  \
0                                              MUCHO                                 
1                                              MUCHO                                 
2                                              MUCHO                                 
3                                              MUCHO                                 
4                                               POCO                                 

  Conoce algún método para denunciar la existencia de baches en su lugar de residencia.  \
0                                                 NO                                      
1                                                 SI                                      
2                                                 SI                                      
3                                                 SI                                      
4                                                 SI                                      

  Puede acceder a información sobre los baches de su lugar de residencia. (Denuncias, ubicación y tareas de reparación de baches.)  \
0                                                 NO                                                                                 
1                                         

In [6]:
# Showing columns.
for i in df.columns: print(i)

Marca temporal
Lugar de residencia.
Si selecciono PARAGUAY como lugar de residencia. Puede indicar el departamento en el que vive actualmente.
Como calificaría el estado de las calles en su lugar de residencia.
Considera que exista un problema de baches en su lugar de residencia.
Puede hacer una estimación de la cantidad de baches en su lugar de residencia.
Tipo de camino en su lugar de residencia.
Cuanto tiempo cree que demora la reparación de baches en su lugar de residencia.
Conoce algún método para denunciar la existencia de baches en su lugar de residencia.
Puede acceder a información sobre los baches de su lugar de residencia. (Denuncias, ubicación y tareas de reparación de baches.)
Usted sufrió accidentes a causa de baches en su lugar de residencia.
Si tiene vehículo. Cuan frecuentemente debe hacer mantenimiento a causa del estado de las calles o baches de su lugar de residencia.
Denunciaría la existencia de baches en su lugar de residencia.
Cree que seria util conocer la ubicac

## Saving questions to a new file.

In [7]:
with open("preguntas.txt", "a") as file:
    for k,v in enumerate(df.columns): 
        file.write(f"{k}) {v} \n \n \n")
    file.close()
    

## Renaming columns

In [7]:
new_names = {
'Marca temporal':'marca_temporal',
'Lugar de residencia.':'lugar_residencia',
'Si selecciono PARAGUAY como lugar de residencia. Puede indicar el departamento en el que vive actualmente.':'lugar_residencia_py',
'Como calificaría el estado de las calles en su lugar de residencia.':'estado_calles',
'Considera que exista un problema de baches en su lugar de residencia.':'problema_baches',
'Puede hacer una estimación de la cantidad de baches en su lugar de residencia.':'estimacion_cantidad',
'Tipo de camino en su lugar de residencia.':'tipo_camino',
'Cuanto tiempo cree que demora la reparación de baches en su lugar de residencia.':'tiempo_reparaciones',
'Conoce algún método para denunciar la existencia de baches en su lugar de residencia.':'como_denunciar',
'Puede acceder a información sobre los baches de su lugar de residencia. (Denuncias, ubicación y tareas de reparación de baches.)':'acceso_informacion',
'Usted sufrió accidentes a causa de baches en su lugar de residencia.':'sufrio_accidentes',
'Si tiene vehículo. Cuan frecuentemente debe hacer mantenimiento a causa del estado de las calles o baches de su lugar de residencia.':'frecuencia_mantenimiento',
'Denunciaría la existencia de baches en su lugar de residencia.':'denunciaria_baches',
'Cree que seria util conocer la ubicación y otros datos de los baches en su lugar de residencia.':'util_conocer',
'Si considera que su lugar de residencia tiene un problema de baches. Como califica la repuesta gubernamental al problema.':'respuesta_gubernamental',
'Puede estimar la regularidad con la que se reparan las calles en su lugar de residencia actual.':'regularidad_reparaciones',
}

In [8]:
df.rename(columns=new_names, inplace=True)

In [9]:
for i in df.columns: 
    print(i)
    print('=============')

marca_temporal
lugar_residencia
lugar_residencia_py
estado_calles
problema_baches
estimacion_cantidad
tipo_camino
tiempo_reparaciones
como_denunciar
acceso_informacion
sufrio_accidentes
frecuencia_mantenimiento
denunciaria_baches
util_conocer
respuesta_gubernamental
regularidad_reparaciones


# General Inspection

In [10]:
df.shape

(439, 16)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 439 entries, 0 to 438
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   marca_temporal            439 non-null    object
 1   lugar_residencia          439 non-null    object
 2   lugar_residencia_py       247 non-null    object
 3   estado_calles             436 non-null    object
 4   problema_baches           437 non-null    object
 5   estimacion_cantidad       435 non-null    object
 6   tipo_camino               438 non-null    object
 7   tiempo_reparaciones       437 non-null    object
 8   como_denunciar            436 non-null    object
 9   acceso_informacion        438 non-null    object
 10  sufrio_accidentes         438 non-null    object
 11  frecuencia_mantenimiento  386 non-null    object
 12  denunciaria_baches        435 non-null    object
 13  util_conocer              436 non-null    object
 14  respuesta_gubernamental   

 - Tenemos 439 respuestas a la encuesta.

## Contamos valores null por columna.

In [12]:
for i in df.columns:
    print(f'Columna: {i} -> {df[i].isna().sum()}')
    

Columna: marca_temporal -> 0
Columna: lugar_residencia -> 0
Columna: lugar_residencia_py -> 192
Columna: estado_calles -> 3
Columna: problema_baches -> 2
Columna: estimacion_cantidad -> 4
Columna: tipo_camino -> 1
Columna: tiempo_reparaciones -> 2
Columna: como_denunciar -> 3
Columna: acceso_informacion -> 1
Columna: sufrio_accidentes -> 1
Columna: frecuencia_mantenimiento -> 53
Columna: denunciaria_baches -> 4
Columna: util_conocer -> 3
Columna: respuesta_gubernamental -> 10
Columna: regularidad_reparaciones -> 5


        En el lugar de residencia en paraguay se presenta la mayor cantidad de valores "null" esto se debe a que la encuesta la respondieron personas de todas partes del mundo.

In [13]:
# paises_mas_uno = lr[lr > 1]

# MOSTRAR CONTEO GENERAL POR COLUMNA

In [17]:
def contar_valores_por_columna():
    for i in df.columns:
        print(f'MOSTRANDO -> {i}')
        print(df[i].value_counts())
        print('\n')

In [27]:
df['respuesta_gubernamental'].replace(
    {'EXELENTE': 'EXCELENTE'}, 
    regex=True, 
    inplace=True)

df['estado_calles'].replace(
    {'Exelente': 'EXCELENTE'}, 
    regex=True, 
    inplace=True)

In [28]:
contar_valores_por_columna()

MOSTRANDO -> marca_temporal
29/10/2022 21:02:13    1
8/11/2022 22:41:21     1
8/11/2022 23:27:59     1
8/11/2022 23:25:49     1
8/11/2022 23:21:49     1
                      ..
3/11/2022 12:27:04     1
3/11/2022 12:00:17     1
3/11/2022 11:56:01     1
3/11/2022 11:52:08     1
9/11/2022 20:46:50     1
Name: marca_temporal, Length: 439, dtype: int64


MOSTRANDO -> lugar_residencia
Paraguay                       236
Mexico                          47
Argentina                       34
Spain                           21
Dominican Republic              19
Peru                            15
Venezuela                       13
Colombia                        12
Bolivia                         12
Chile                            5
Panama                           2
Honduras                         2
Uruguay                          2
Anguilla                         2
Guatemala                        2
Italy                            1
Cuba                             1
Papua New Guinea      

In [29]:
df.head()

marca_temporal lugar_residencia lugar_residencia_py estado_calles  \
0  29/10/2022 21:02:13        Venezuela                 NaN          Malo   
1   30/10/2022 9:28:12        Argentina                 NaN         Bueno   
2  30/10/2022 10:05:24           Mexico                 NaN    Lamentable   
3  30/10/2022 10:12:40            Spain                 NaN         Bueno   
4  30/10/2022 10:21:56        Singapore             Central     EXCELENTE   

  problema_baches estimacion_cantidad tipo_camino tiempo_reparaciones  \
0              SI              MUCHOS     ASFALTO               MUCHO   
1              SI               POCOS     ASFALTO               MUCHO   
2              SI              MUCHOS     ASFALTO               MUCHO   
3              SI               POCOS     ASFALTO               MUCHO   
4              NO                 NaN     ASFALTO                POCO   

  como_denunciar acceso_informacion sufrio_accidentes  \
0             NO                 NO                SI   
1             SI                 NO                NO   
2             SI                 SI                SI   
3             SI                 NO                NO   
4             SI                 SI          NO NUNCA   

  frecuencia_mantenimiento            denunciaria_baches util_conocer  \
0             REGULARMENTE                            SI           SI   
1           FRECUENTEMENTE                            SI           SI   
2             REGULARMENTE  SI NO ME LLEVA MUCHO TIEMPO.           SI   
3             REGULARMENTE                            SI           SI   
4           FRECUENTEMENTE                            SI           NO   

  respuesta_gubernamental regularidad_reparaciones  
0              LAMENTABLE                    NUNCA  
1              LAMENTABLE             REGULARMENTE  
2              LAMENTABLE             REGULARMENTE  
3                   BUENA             REGULARMENTE  
4               EXCELENTE             REGULARMENTE

In [30]:
df.to_csv('encuesta_baches_fixed.csv')